tests/test_collection_info.py

```diff
@@ -0,0 +1,302 @@
+    assert res.license_file == 'my_very_own_license.txt'
+
+
+def test_empty_lic_and_lic_file(collection_info):
+    collection_info['license'] = []
+    with pytest.raises(ValueError,
+                       match=r"'license' or 'license_file' are required"):
+        CollectionInfo(**collection_info)
+
+
+def test_license_not_list_of_str(collection_info):
+    collection_info['license'] = 'MIT'
+    with pytest.raises(ValueError, match=r"'license' to be a list of strings"):
+        CollectionInfo(**collection_info)
+
+
+@pytest.mark.parametrize(
+    'valid_license_list',
+    [
+        ['MIT'],
+        ['Apache-2.0'],
+        ['BSD-3-Clause'],
+        ['CC-BY-4.0', 'MIT']
+    ]
+)
+def test_valid_license(collection_info, valid_license_list):
+    collection_info['license'] = valid_license_list
+    res = CollectionInfo(**collection_info)
+    assert res.license == valid_license_list
+
+
+@pytest.mark.parametrize(
+    'invalid_license_list',
+    [
+        ['BSD'],
+        ['Apache'],
+        ['MIT AND Apache-2.0'],
+        ['something_else'],
+        ['CC-BY-4.0', 'MIT', 'bad_license_id']
+    ]
+)
+def test_invalid_license(collection_info, invalid_license_list):
+    collection_info['license'] = invalid_license_list
+    with pytest.raises(ValueError, match=r'invalid license identifiers'):
+        CollectionInfo(**collection_info)
+
+
+def test_invalid_dep_type(collection_info):
+    collection_info['dependencies'] = 'joe.role1: 3'
+    with pytest.raises(TypeError,
+                       match=r"'dependencies' must be <class 'dict'>"):
+        CollectionInfo(**collection_info)
+
+
+def test_invalid_dep_name(collection_info):
+    collection_info['dependencies'] = {3.3: '1.0.0'}
+    with pytest.raises(ValueError, match=r'Expecting depencency to be string'):
+        CollectionInfo(**collection_info)
+
+
+def test_invalid_dep_version(collection_info):
+    collection_info['dependencies'] = {'joe.role1': 3}
+    with pytest.raises(ValueError,
+                       match=r'Expecting depencency version to be string'):
+        CollectionInfo(**collection_info)
+
+
+def test_non_null_str_fields(collection_info):
+    collection_info['description'] = None
+    res = CollectionInfo(**collection_info)
+    assert res.description is None
+
+    collection_info['description'] = 'description of the collection'
+    res = CollectionInfo(**collection_info)
+    assert res.description == 'description of the collection'
+
+    collection_info['description'] = ['should be a string not list']
+    with pytest.raises(ValueError, match=r"'description' must be a string"):
+        CollectionInfo(**collection_info)
+
+
+@pytest.mark.parametrize(
+    'dependent_collection',
+    [
+        'no_dot_in_collection',
+        'too.many.dots',
+        '.too.many.dots',
+        'too.many.dots.',
+        'empty_name.',
+        '.empty_namespace',
+        'a_user._leading_underscore',
+        '5tarts_with_number.gunicorn',
```

Comment:
Does not test for spaces, dashes, or uppercase letters, all of which are invalid but are not caught when I try to pass them.

Reply:
I added those as tests. They are caught as expected, @ironfroggy what are the values that you are using?